File: etl-script-v2/src/useCases/etl.ts
import Fetcher from "src/ports/Fetcher";
import Saver from "src/ports/Saver";

export default class ETL {

    constructor(private fetcher: Fetcher, private saver: Saver) {

    }

    exec(): void{
        this.fetcher.fetch()
            .then(result => this.saver.save(result))
            .then(result => console.log(result))
            .catch(err => console.error(err));
    }


}

File: java/tech_start/src/main/java/guru/orlando/tech_start/usecases/ETL.java
package guru.orlando.tech_start.usecases;

import java.util.List;

import guru.orlando.tech_start.dtos.ResultDTO;
import guru.orlando.tech_start.ports.Fetcher;
import guru.orlando.tech_start.ports.Saver;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@AllArgsConstructor
@Slf4j
public class ETL {
	
	private Fetcher fetcher;
	private Saver saver;
	
	public void exec(){
		List<ResultDTO> results = fetcher.fetch();
		saver.save(results);
		log.info("COMPLETED!");
	}
	
}
